### DISCUSSION TITLE
Questions regarding the rewetting options in modflow 6

### DISCUSSION THREAD

**Discussion by sebmeng** [modflowpy/flopy#2143] [Q&A] [answered]
Hello Community!
I have a question regarding the rewetting options in modflow 6 (using flopy 3.6.0). If I create my transient model with icelltype=0 and iconvert=0 (all layers confined), it runs fine. But from the conceptual model I know that at least my first layer (of four layers in total) is phreatic. 
If I create then the npf and sto packages like this: 
`# Create new NPF and STO packages for the current model using the parameters from the source model
npf = flopy.mf6.ModflowGwfnpf(gwf, pname='npf', save_flows=source_npf.save_flows, 
                              icelltype=[1,0,0,0], k=source_npf.k.array, rewet_record=[("WETFCT", 1.1, "IWETIT", 0, "IHDWET", 1)], wetdry=-0.0001)
sto = flopy.mf6.ModflowGwfsto(gwf, pname='sto', save_flows=source_sto.save_flows, 
                              iconvert=[1,0,0,0], ss=source_sto.ss.array, sy=source_sto.sy.array, transient=True) #only first layer should be convertible
`
the result is that some cells gets dry (head values of -1e30, even if i setted the drywet parameter):
![image](https://github.com/modflowpy/flopy/assets/48210348/47ff65e6-8598-47f8-943c-cf492ff8b56e)

I have tried moving the values of wetdry, and wetfct but I am getting the same results, can someone help me to know where I'm making a mistake?

thanks in advance for any help or guidance,

Regards!

Sebastian

**Comment by javgs-bd** [comment, marked as answer]
Hi Sebastian
I'm not answering your original question (apologies) but I wonder, if you do know you have phreatic conditions maybe makes sense to use the  Newton-raphson and under-relaxation instead of rewetting options. See #1997 for some details that might help. 

Javier

**Reply by sebmeng** [reply]
Hello Javier
Thank you for the idea! It worked just fine! Creating the model with icelltype and iconvert = 1 and using similar values to the ones you suggested for the ims in #1997 :
```
# Create the groundwater flow model object
gwf = flopy.mf6.ModflowGwf(sim, modelname=modelname, save_flows=True,newtonoptions="UNDER_RELAXATION")


# Create the iterative model solver object
ims = flopy.mf6.ModflowIms(
    sim,
    print_option='SUMMARY',
    outer_dvclose=1e-5,
    outer_maximum=600,
    under_relaxation='DBD',
    under_relaxation_gamma          = 0.003,
    under_relaxation_theta          = 0.8,
    under_relaxation_kappa          = 0.001,
    under_relaxation_momentum       = 0.0,
    backtracking_number             = 200,
    backtracking_tolerance          = 1.5, 
    backtracking_reduction_factor   = 0.9,
    backtracking_residual_limit     = 1,
    inner_maximum=600,
    inner_dvclose=1e-5,
    rcloserecord=[1e-5, 'RELATIVE_RCLOSE'],
    linear_acceleration='BICGSTAB',
    relaxation_factor=0.97,
    preconditioner_levels=7,
    preconditioner_drop_tolerance=1e-7,
    number_orthogonalizations=10,
    scaling_method='NONE',
    reordering_method='NONE',
)
# Create new NPF and STO packages for the current model using the parameters from the source model
npf = flopy.mf6.ModflowGwfnpf(gwf, pname='npf', save_flows=source_npf.save_flows, 
                              icelltype=[1,0,0,0], k=source_npf.k.array)
sto = flopy.mf6.ModflowGwfsto(gwf, pname='sto', save_flows=source_sto.save_flows, 
                              iconvert=[1,0,0,0], ss=source_sto.ss.array, sy=source_sto.sy.array, transient=True) #only first layer should be convertible


```
The model runs without dry cells. Now the simulation time increased, so I will implement the ATS sccheme that before wasn't include,

Thanks again!
Sebastian

**Comment by javgs-bd** [comment]
Glad to hear. Before you implement ATS I would recommend to play around with the gamma and kappa parameters. They are extremely low in that setup. That might be cause of the long runtime.
Also try to relax the backtracking parameters and play with the preconditioner options too. You can find guidance in the flopy docs for IMS. 

Look into the LST files to know if the inner or outer solver are having problems.

Hope it helps

**Reply by sebmeng** [reply]
Great idea Javier, I will do it before going with ATS.
Thanks for the help!

Sebastian
